pkg/webhook/mutation/pod/metadata/workload.go

```diff
@@ -46,43 +39,29 @@ func findRootOwnerOfPod(ctx context.Context, clt client.Client, pod *corev1.Pod,
 			Kind:       pod.Kind,
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: pod.ObjectMeta.Name,
+			Name: kubeobjects.GetName(*pod),
 			// pod.ObjectMeta.Namespace is empty yet
 			Namespace:       namespace,
 			OwnerReferences: pod.ObjectMeta.OwnerReferences,
 		},
 	}
 
-	workloadInfo, err := findRootOwner(ctx, clt, podPartialMetadata)
+	workloadInfo, err := findRootOwner(ctx, clt, podPartialMetadata, podPartialMetadata) // default owner of the pod is the pod itself
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
 	return &workloadInfo, nil
 }
 
-func findRootOwner(ctx context.Context, clt client.Client, partialObjectMetadata *metav1.PartialObjectMetadata) (workloadInfo, error) {
-	if len(partialObjectMetadata.ObjectMeta.OwnerReferences) == 0 {
-		if partialObjectMetadata.ObjectMeta.Name == "" {
-			// pod is not created directly and does not have an owner reference set
-			return newUnknownWorkloadInfo(), nil
-		}
-
-		return newWorkloadInfo(partialObjectMetadata), nil
+func findRootOwner(ctx context.Context, clt client.Client, rootObject *metav1.PartialObjectMetadata, wellKnownOwner *metav1.PartialObjectMetadata) (workloadInfo, error) {
+	if len(rootObject.ObjectMeta.OwnerReferences) == 0 {
+		return newWorkloadInfo(wellKnownOwner), nil
```

Comment:
IMO we should only create the `workloadInfo` outside of `findRootOwner`, so we work with only `*metav1.PartialObjectMetadata` making things more straight forward